src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,324 @@
+
+## Model description
+More information needed
+
+## Intended uses & limitations
+More information needed
+
+## Training and evaluation data
+More information needed
+
+
+
+"""
+
+# Define template for a auto-generated config with fastai and fastcore versions
+CONFIG_TEMPLATE = dict(
+    fastai_version=get_fastai_version(), fastcore_version=get_fastcore_version()
+)
+
+
+def _create_model_card(repo_dir: Path):
+    """
+    Creates a model card for the repository.
+
+    Parameters:
+        repo_dir (:obj:`Path`):
+            Directory where model card is created.
+    """
+    readme_path = repo_dir / "README.md"
+    readme = ""
+    if readme_path.exists():
+        with readme_path.open("r", encoding="utf8") as f:
+            readme = f.read()
+    else:
+        readme = README_TEMPLATE
+    with readme_path.open("w", encoding="utf-8") as f:
+        f.write(readme)
+
+
+def save_fastai_learner(
+    learner,
+    save_directory: str,
+    config: Optional[Dict[str, Any]] = None,
+    pickle_protocol: Optional[int] = 2,
+):
+    """
+    Saves a fastai learner to save_directory in pickle format. Use this if you're using Learners.
+
+    Parameters:
+        learner (:obj:`Learner`):
+            The `fastai.Learner` you'd like to save.
+        save_directory (:obj:`str`):
+            Specific directory in which you want to save the fastai learner.
+        config (:obj:`dict`, `optional`):
+            Configuration object. Will be uploaded as a .json file. Example: 'https://huggingface.co/espejelomar/fastai-pet-breeds-classification/blob/main/config.json'.
+        pickle_protocol (:obj:`int`, `optional`):
+            Pickle protocol passed to torch.save. Refer to pickle documentation.
+    """
+    # Check that fastai and fastcore versions are supported.
+    check_fastai_fastcore_versions()
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # saving config
+    # if user provides config then we update it with the fastai and fastcore versions in CONFIG_TEMPLATE.
+    if config is not None:
+        if not isinstance(config, dict):
+            raise RuntimeError(
+                f"Provided config should be a dict. Got: '{type(config)}'"
+            )
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump({**config, **CONFIG_TEMPLATE}, f)
+    else:
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump(CONFIG_TEMPLATE, f)
+
+    # creating README.md if none exist
+    _create_model_card(Path(save_directory))
+
+    # saving learner
+    # learner.export saves the model in `self.path/save_directory` and this folder should exist.
+    # We create this folder and call it `fastai_path`
+    fastai_path = os.path.join(learner.path, save_directory)
+    os.makedirs(fastai_path, exist_ok=True)
+    learner.export(
+        fname=os.path.join(save_directory, "model.pkl"), pickle_protocol=pickle_protocol
```

Comment:
Is specifying different pickle protocols something that will be needed @muellerzr ?

Reply:
Thinking on it more, it *shouldn't* be. (Talked with Jeremy about this a moment ago as well). Reason being if not passed in, it should pull the default for Python.

This is still a fix that needs to be done in fastai, so proposed solution is to set it to `None` when being passed to `export` so python takes over.

Reply:
Or, even better: `pickle.DEFAULT_PROTOCOL`

Reply:
Will use pickle.DEFAULT_PROTOCOL! Thanks for the suggetion :D